.github/generate-release-notes.py

```diff
@@ -0,0 +1,341 @@
+"""Automatically generate release notes from GitHub issues and PRs.
+
+Run `uv run .github/generate-release-notes.py` to generate the release notes.
+"""
+# /// script
+# dependencies = [
+#   "rich",
+#   "PyGithub",
+#   "diskcache",
+#   "GitPython",
+# ]
+# ///
```

Comment:
@evelyn-king, this is indeed a super cool feature!

And since you asked for CHANGELOGs, here's an attempt 😄